jax/_src/lax/parallel.py

```diff
@@ -1478,26 +1478,31 @@ def psum_scatter(x, axis_name, *, scatter_dimension=0, axis_index_groups=None, t
 
```

Comment:
I have to say this description isn't good enough to communicate the intuition, and it's probably better to get the intuition before reading the details...

Reply:
You're right. Maybe like "Compute a ``psum``, but retain only part of the result on each device."? Or something more precise?

Reply:
For precision, we could add that `psum_scatter(x, axis_name, scatter_dimension=0, tiled=False)` produces the same value as `psum(x, axis_name)[jax.lax.axis_index(axis_name)]`, but more efficient. But that doesn't fit in the first line of the docstring.

Reply:
<img width="572" alt="image" src="https://github.com/google/jax/assets/1458824/4cf777e1-e4ee-46e5-a59a-2845bf5df056">